fuzz/corpus_test.go

```diff
@@ -0,0 +1,122 @@
+package fuzz
+
+import (
+	"context"
+	"github.com/stretchr/testify/assert"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func listFilesRecursively(dir string) ([]string, error) {
+	var files []string
+
+	if err := filepath.WalkDir(dir, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if d.IsDir() {
+			return nil
+		}
+		rel, err := filepath.Rel(dir, path)
+		if err != nil {
+			return err
+		}
+		files = append(files, rel)
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+
+	return files, nil
+}
+
+func TestCorpusExtract(t *testing.T) {
+	t.Run("extracts fuzzing targets", func(t *testing.T) {
+		project := Project{Directory: "./testdata/corpus/multiple"}
+		ctx := context.Background()
+
+		tempDir, err := os.MkdirTemp("", "go-ci-fuzz-*")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		t.Cleanup(func() {
+			if os.RemoveAll(tempDir) != nil {
+				t.Error(err)
+			}
+		})
```

Comment:
Here we should be able to avoid all the temp dir creation and code clean up as it is already supported by the testing  package with https://pkg.go.dev/testing#T.TempDir (I love it 😍 )